src/view/components/admin-sections/movies/movieCRUD.php

```diff
@@ -98,500 +162,622 @@
-            alert('Success! Movie archived successfully.');
-            window.location.reload();
-            clearValues('archive');
-            archiveMovieModal.classList.add('hidden');
-        } else {
-            console.error('Error:', response.errorMessage);
+        // Close the restore modal
+        if (cancelRestoreMovieButton) {
+            cancelRestoreMovieButton.addEventListener('click', () => {
+                restoreMovieModal.classList.add('hidden');
+                restoreMovieIdInput.value = '';
+            });
         }
-    }
-};
-    // Send the request with movieId in URL parameters or form data
-    xhr.send(`movieId=${movieId}&action=archiveMovie`);
-});
-
-    //Clear error messages and input values
-    function clearValues(action) {
-        if (action === 'edit') {
-        editMovieForm.reset();
-        } else if (action === 'add') {
-            addMovieForm.reset();
-        } else if (action === 'archive') {
-            errorArchiveMovieMessageGeneral.classList.add('hidden');
-            archiveMovieIdInput.value = '';
-            archiveMovieForm.reset();
+
+        // Confirm restore movie
+        confirmRestoreMovieButton.addEventListener('click', function(event) {
+            const xhr = new XMLHttpRequest();
+            const baseRoute = '<?php echo $_SESSION['baseRoute']; ?>';
+            const movieId = restoreMovieIdInput.value;
+
+            xhr.open('PUT', `${baseRoute}movies/restore`, true);
+
+            xhr.onreadystatechange = function() {
+                if (xhr.readyState === 4 && xhr.status === 200) {
+                    let response;
+                    try {
+                        response = JSON.parse(xhr.response);
+                    } catch (e) {
+                        console.error('Could not parse response as JSON:', e);
+                        return;
+                    }
+
+                    if (response.success) {
+                        alert('Success! Movie restored successfully.');
+                        window.location.reload();
+                        restoreMovieModal.classList.add('hidden');
+                    } else {
+                        console.error('Error:', response.errorMessage);
+                    }
+                }
+            };
+            // Send the request with movieId in URL parameters or form data
+            xhr.send(`movieId=${movieId}&action=restoreMovie`);
+        });
+
+        //Clear error messages and input values
+        function clearValues(action) {
+            if (action === 'edit') {
+                editMovieForm.reset();
+            } else if (action === 'add') {
+                addMovieForm.reset();
+            } else if (action === 'archive') {
+                errorArchiveMovieMessageGeneral.classList.add('hidden');
+                archiveMovieIdInput.value = '';
+                archiveMovieForm.reset();
+            }
         }
-    }
-});
-</script>
+    });
+</script>
```

Comment:
as we talked about it, it's an incredibely huge file :D maybe worth to try to separate it